src/charm.py

```diff
@@ -3,162 +3,272 @@
+                        "override": "replace",
+                        "period": "30s",
+                        "http": {"url": "http://localhost:8080/metrics"},
                     },
-                ],
-            },
-            k8s_resources={
-                "kubernetesResources": {
-                    "customResourceDefinitions": [
-                        {"name": crd["metadata"]["name"], "spec": crd["spec"]}
-                        for crd in yaml.safe_load_all(Path("files/crds.yaml").read_text())
-                    ],
                 },
-                "configMaps": {
-                    "namespace-labels-data": {
-                        namespace_labels_filename: Path(
-                            f"files/{namespace_labels_filename}"
-                        ).read_text(),
+            }
+        )
+
+    @property
+    def _kfam_pebble_layer(self) -> Layer:
+        """Return the Pebble layer for the workload."""
+        return Layer(
+            {
+                "services": {
+                    self._kfam_container_name: {
+                        "override": "replace",
+                        "summary": "entry point for kubeflow access management",
+                        "command": (
+                            "/access-management "
+                            "-cluster-admin "
+                            "admin "
+                            "-userid-header "
+                            "kubeflow-userid "
+                            "-userid-prefix "
+                            '""'
+                        ),
+                        "startup": "enabled",
                     }
                 },
-            },
+                "checks": {
+                    "kubeflow-kfam-get": {
+                        "override": "replace",
+                        "period": "30s",
+                        "http": {"url": "http://localhost:8081/metrics"},
+                    },
+                },
+            }
         )
-        self.log.info("pod.set_spec() completed without errors")
 
-        self.model.unit.status = ActiveStatus()
+    def _deploy_k8s_resources(self):
+        try:
+            self.unit.status = MaintenanceStatus("Creating K8S resources")
+            self.k8s_resource_handler.apply()
+
+        except ApiError:
+            raise ErrorWithStatus("K8S resources creation failed", BlockedStatus)
+        self.model.unit.status = MaintenanceStatus("K8S resources created")
+
+    def _update_profiles_layer(self) -> None:
+        """Updates the Pebble configuration layer if changed."""
+        if not self.profiles_container.can_connect():
+            raise ErrorWithStatus("Waiting for pod startup to complete", MaintenanceStatus)
+
+        current_layer = self.profiles_container.get_plan()
+
+        if current_layer.services != self._profiles_pebble_layer.services:
+            with open("src/files/namespace-labels.yaml", encoding="utf-8") as labels_file:
```

Comment:
Should we define filenames via constants?
I.e. NAMESPACE_LABELS_FILE="src/files/namespace-labels.yaml"